server/routes/journeys/update-investigation/triggers/controller.ts

```diff
@@ -0,0 +1,35 @@
+import { NextFunction, Request, Response } from 'express'
+import { SchemaType } from '../../record-investigation/triggers/schemas'
+import { MESSAGE_INVESTIGATION_UPDATED, PatchInvestigationController } from '../../base/patchInvestigationController'
+import { generateSaveTimestamp, getMaxCharsAndThresholdForAppend } from '../../../../utils/appendFieldUtils'
+
+export class UpdateTriggersController extends PatchInvestigationController {
+  GET = async (req: Request, res: Response) => {
+    const currentPersonsTrigger = req.journeyData.csipRecord!.referral.investigation!.personsTrigger
+    res.render('record-investigation/triggers/view', {
+      currentPersonsTrigger,
+      personsTrigger: res.locals.formResponses?.['personsTrigger'],
+      isUpdate: true,
+      recordUuid: req.journeyData.csipRecord!.recordUuid,
+      ...getMaxCharsAndThresholdForAppend(res.locals.user.displayName, currentPersonsTrigger),
+    })
+  }
+
+  checkSubmitToAPI = async (req: Request<unknown, unknown, SchemaType>, res: Response, next: NextFunction) =>
+    this.submitChanges({
+      req,
+      res,
+      next,
+      changes: {
+        personsTrigger:
+          (req.journeyData.investigation!.personsTrigger ?? '') +
+          generateSaveTimestamp(res.locals.user.displayName) +
+          req.body.personsTrigger,
```

Comment:
appendFieldUtils has a helper function for this `getTextForApiSubmission`

Reply:
Updated this and other controllers where we could simplify and use this function